src/nix/search.cc

```diff
@@ -117,7 +117,16 @@ struct CmdSearch : InstallableValueCommand, MixJSON
                     auto aDescription = aMeta ? aMeta->maybeGetAttr(state->sDescription) : nullptr;
                     auto description = aDescription ? aDescription->getString() : "";
                     std::replace(description.begin(), description.end(), '\n', ' ');
-                    auto attrPath2 = concatStringsSep(".", attrPathS);
+                    std::vector<std::string> attrPathS2;
+                    for (auto&& elem : attrPathS) {
+                        if (!(
+                            elem == "legacyPackages"
+                            || elem == "packages"
+                            || elem == settings.thisSystem.get())) {
+                            attrPathS2.push_back(elem);
+                        }
+                    }
+                    auto attrPath2 = concatStringsSep(".", attrPathS2);
```

Comment:
I think the suggestion from the Nix Team basically means that you should replace all this logic with something akin to (don't know of the top of my head if `last` is the correct name):

```suggestion
                    auto attrPath2 = attrPathS.last();
```

@tomberek can you confirm that is what you meant?

Reply:
I think C++ calls it `back()`.

I was in the meeting and my understanding of it matches what @iFreilicht describes.

For context, we considered alternatives, but those were complicated, and we expect this simpler solution to be a better trade-off in terms of implementation complexity and effectiveness, at least for the time being.

Reply:
Correct. This would be a change of behavior., but may be ultimately simpler than having exceptions and surprising behavior. Thoughts?

Reply:
Confirmed today that the above suggestion of using only the last[back] element of `attrPathS` is the best path forward.